roxie/udplib/udptrr.cpp

```diff
@@ -806,20 +1441,28 @@ class CReceiveManager : implements IReceiveManager, public CInterface
     uid_map         collators;
     CriticalSection collatorsLock; // protects access to collators map
 
-  public:
+public:
     IMPLEMENT_IINTERFACE;
-    CReceiveManager(int server_flow_port, int d_port, int client_flow_port, int queue_size, int m_slot_pr_client, bool _encrypted)
-        : collatorThread(*this), encrypted(_encrypted), sendersTable([client_flow_port](const ServerIdentifier ip) { return new UdpSenderEntry(ip.getIpAddress(), client_flow_port);})
+    CReceiveManager(int server_flow_port, int d_port, int client_flow_port, int queue_size, bool _encrypted)
+        : collatorThread(*this), encrypted(_encrypted),
+        sendersTable([client_flow_port](const ServerIdentifier ip) { return new UdpSenderEntry(ip.getIpAddress(), client_flow_port);}),
+        input_queue_size(queue_size), receive_flow_port(server_flow_port), data_port(d_port)
     {
 #ifndef _WIN32
         setpriority(PRIO_PROCESS, 0, -15);
 #endif
-        receive_flow_port = server_flow_port;
-        data_port = d_port;
-        input_queue_size = queue_size;
+        assertex(data_port != receive_flow_port);
         input_queue = new queue_t(queue_size);
         data = new receive_data(*this);
-        receive_flow = new receive_receive_flow(*this, server_flow_port, m_slot_pr_client);
+
+        //NOTE: If all slots are allocated to a single client, then if that server goes down it will prevent any data being received from
+        //any other sender for the udpPermitTimeout period
+        unsigned maxSlotsPerClient = (udpMaxPendingPermits == 1) ? queue_size : (udpMaxClientPercent * queue_size) / (udpMaxPendingPermits * 100);
+        if (maxSlotsPerClient > queue_size)
+            maxSlotsPerClient = queue_size;
+        if (udpResendLostPackets && maxSlotsPerClient > TRACKER_BITS)
+            maxSlotsPerClient = TRACKER_BITS;
```

Comment:
Should we confirm maxSlotsPerClient minimum ?

Reply:
I have added a check that it is non zero (and guarded against it in the sanity checking code).  Is that what you were pointing out?